samcli/lib/providers/sam_api_provider.py

```diff
@@ -98,6 +115,100 @@ def _extract_from_serverless_api(
         collector.stage_variables = stage_variables
         collector.cors = cors
 
+        auth = properties.get(SamApiProvider._AUTH, {})
+        if auth:
+            if auth.get(SamApiProvider._DEFAULT_AUTHORIZER):
+                collector.set_default_authorizer(logical_id, auth.get(SamApiProvider._DEFAULT_AUTHORIZER))
+
+            self._extract_authorizers_from_props(logical_id, auth, collector, Route.API)
```

Comment:
nit: early exit might improve the readability (getting rid of one of the nested if statements). There is also one below with very similar logic, we can apply there as well if we decide to.
```suggestion
        auth = properties.get(SamApiProvider._AUTH, {})
        if not auth:
            return
        
        if auth.get(SamApiProvider._DEFAULT_AUTHORIZER):
            collector.set_default_authorizer(logical_id, auth.get(SamApiProvider._DEFAULT_AUTHORIZER))

        self._extract_authorizers_from_props(logical_id, auth, collector, Route.API)
```